source/simulator/core.cc

```diff
@@ -1042,17 +1042,38 @@ namespace aspect
                mpi_communicator);
 #endif
 
-    DoFTools::make_sparsity_pattern (dof_handler,
-                                     coupling, sp,
-                                     constraints, false,
-                                     Utilities::MPI::
-                                     this_mpi_process(mpi_communicator));
     if ((parameters.use_discontinuous_temperature_discretization) || (parameters.use_discontinuous_composition_discretization))
-      DoFTools::make_flux_sparsity_pattern (dof_handler,
-                                            sp,
-                                            constraints, false,
-                                            Utilities::MPI::
-                                            this_mpi_process(mpi_communicator));
+      {
+        Table<2,DoFTools::Coupling> face_coupling (introspection.n_components,
+                                                   introspection.n_components);
+        const typename Introspection<dim>::ComponentIndices &x
+          = introspection.component_indices;
+        if (parameters.use_discontinuous_temperature_discretization)
+          face_coupling[x.temperature][x.temperature] = DoFTools::always;
+
+        if (parameters.use_discontinuous_composition_discretization)
+          {
+            for (unsigned int c=0; c<parameters.n_compositional_fields; ++c)
+              face_coupling[x.compositional_fields[c]][x.compositional_fields[c]] = DoFTools::always;
+          }
+        /*
+                DoFTools::make_sparsity_pattern (dof_handler,
+                                               coupling, sp,
+                                               constraints, false,
+                                               Utilities::MPI::
+                                               this_mpi_process(mpi_communicator));
+        */
+        DoFTools::make_flux_sparsity_pattern (dof_handler,
+                                              sp,
+                                              coupling,
+                                              face_coupling);
```

Comment:
@bangerth , @tjhei ,  it seems that we need a new function which combines  the following two functions:

One to handle the parallel and constraints
void DoFTools::make_flux_sparsity_pattern	(	const DoFHandlerType & 	dof_handler,
SparsityPatternType & 	sparsity_pattern,
const ConstraintMatrix & 	constraints,
const bool 	keep_constrained_dofs = true,
const types::subdomain_id 	subdomain_id = numbers::invalid_unsigned_int 
)	

One to handle the coupling of different fields between FEM and DG
void DoFTools::make_flux_sparsity_pattern	(	const DoFHandlerType & 	dof,
SparsityPatternType & 	sparsity,
const Table< 2, Coupling > & 	cell_integrals_mask,
const Table< 2, Coupling > & 	face_integrals_mask 
)	

Unfortunately, we don't have such one in deal.ii satisfying those two features. So if we have to extend the make_flux_sparsity_pattern function, any suggestion? Or if there is something we can avoid to do that, it will be best...